### PR TITLE
Allow rpc expose definition/ metadata

### LINKED ISSUE #3494: Can not access Rpc definition/ metadata
It appears impossible for the user to retrieve the RPC method's metadata/ definition, as it currently not been exposed when decorating the RpcCore . 

Some useful information is missing, such as params (type,isHistoric, isOptional etc). And it will be more difficult for users who want to use customized rpc and types.

### PR DESCRIPTION
**Pull request by jiqiang90**
Closes #3494

This PR allow user to call `api.rpc.section.method.def` and `api.rx.rpc.section.method.def` to get rpc's definition (metadata).

### CODE REVIEW

File: packages/api/src/types/rpc.ts [hunk 1]
```diff
@@ -11,13 +11,15 @@ import { ApiTypes, PromiseResult, Push, RxResult, UnsubscribePromise } from './b
 export interface RpcRxResult<F extends AnyFunction> extends RxResult<F> {
   json (...args: Parameters<F>): Observable<Json>;
   raw (...args: Parameters<F>): Observable<Raw>;
+  def: DefinitionRpc;
```
Comment (jacogr) [inline packages/api/src/types/rpc.ts:14]: Just one small comment - would adjust this to be `meta`, just to align with what we have in the rest for this type of info.

File: packages/api/test/api.spec.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ function createApi (): Promise<ApiPromise> {
   return new ApiPromise({ provider }).isReady;
 }
 
-describe.skip('misc quick tests', (): void => {
+describe.only('misc quick tests', (): void => {
```
Comment (jacogr) [inline packages/api/test/api.spec.ts:17]:
```suggestion
describe('misc quick tests', (): void => {
```

File: packages/api/test/api.spec.ts [hunk 3]
```diff
@@ -76,4 +76,12 @@ describe.skip('misc quick tests', (): void => {
         .map(([block, value]) => [block, value.toRawType(), value.toHuman()])
     ));
   });
+
+  it.only('expose rpc and rx definition', async (): Promise<void> => {
```
Comment (jacogr) [inline packages/api/test/api.spec.ts:80]:
```suggestion
  it('expose rpc and rx definition', async (): Promise<void> => {
```

File: packages/rpc-core/src/types.ts [hunk 4]
```diff
@@ -3,10 +3,13 @@
...
 
+import { DefinitionRpc } from '@polkadot/types/types';
+
 export * from './types.jsonrpc';
 
 export interface RpcInterfaceMethod {
   (...params: any[]): Observable<any>;
   json (...params: any[]): Observable<any>;
   raw (...params: any[]): Observable<any>;
+  def: DefinitionRpc;
```
Comment (jacogr) [inline packages/rpc-core/src/types.ts:14]:
```suggestion
  meta: DefinitionRpc;
```

File: packages/api/src/types/rpc.ts (hunk 1 above)
Comment (jacogr) [inline packages/api/src/types/rpc.ts:14]:
```suggestion
  meta: DefinitionRpc;
```

File: packages/api/src/types/rpc.ts [hunk 5]
```diff
@@ -11,13 +11,15 @@ import { ApiTypes, PromiseResult, Push, RxResult, UnsubscribePromise } from './b
...
 }
 
 export interface RpcPromiseResult<F extends AnyFunction> extends PromiseResult<F> {
   json (...args: Parameters<F>): Promise<Json>;
   json (...args: Push<Parameters<F>, Callback<Json>>): UnsubscribePromise;
   raw (...args: Parameters<F>): Promise<Raw>;
   raw (...args: Push<Parameters<F>, Callback<Raw>>): UnsubscribePromise;
+  def: DefinitionRpc;
```
Comment (jacogr) [inline packages/api/src/types/rpc.ts:22]:
```suggestion
  meta: DefinitionRpc;
```

File: packages/api/src/base/Decorate.ts [hunk 6]
```diff
@@ -297,6 +297,8 @@ export abstract class Decorate<ApiType extends ApiTypes> extends Events {
             (section as Record<string, { json: unknown }>)[methodName].json = decorateMethod(method.json, { methodName }) as unknown;
             // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
             (section as Record<string, { raw: unknown }>)[methodName].raw = decorateMethod(method.raw, { methodName }) as unknown;
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            (section as Record<string, { def: unknown }>)[methodName].def = method.def;
```
Comment (jacogr) [inline packages/api/src/base/Decorate.ts:301]:
```suggestion
            (section as Record<string, { meta: unknown }>)[methodName].meta = method.def;
```

**Review by jacogr — COMMENTED** [review]
Thank you. This is certainly a very useful addition. Just made a couple of small comments.

File: packages/api/src/types/rpc.ts (hunk 1 above)
Comment (jiqiang90) [reply packages/api/src/types/rpc.ts]: Thanks Jaco. Change has been made, can you please check again.

**Review by jacogr — APPROVED** [review]
Thank you very much. This is a nice addition.

**Pull request merged**